sdk/test/UnitTests/Custom/Marshalling/StreamingUtf8JsonReaderTests.cs

```diff
@@ -0,0 +1,229 @@
+﻿using AWSSDK_DotNet.CommonTest.Utils;
+using AWSSDK_DotNet.UnitTests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Amazon.Runtime.Internal.Util;
+using Amazon.EC2.Model;
+using Amazon;
+namespace AWSSDK.UnitTests
+{
+    /// <summary>
+    /// Protocol tests already exists to test the marshalling and unmarshalling of request and responses in json, but they don't test very
+    /// large payloads, which would trigger the logic for getting more bytes from the stream. 
+    /// This class just tests the wrapper class StreamingUtf8JsonReader. 
+    /// </summary>
+    [TestClass]
+    public class StreamingUtf8JsonReaderTests
+    {
+        private static int originalBufferSize;
+        [ClassInitialize]
+        public static void Initialize(TestContext testContext)
+        {
+            originalBufferSize = AWSConfigs.StreamingUtf8JsonReaderBufferSize.GetValueOrDefault();
+            AWSConfigs.StreamingUtf8JsonReaderBufferSize = 4096;
```

Comment:
I would recommend adding a second constructor on the `StreamingUtf8JsonReader` that takes in the buffer size so you don't have to toggle the `AWSConfigs.StreamingUtf8JsonReaderBufferSize` global setting.